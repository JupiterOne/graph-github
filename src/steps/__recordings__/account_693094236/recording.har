{
  "log": {
    "_recordingName": "account",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "f13f3f3dcf87065d27af1cac191fa6e2",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/vnd.github.machine-man-preview+json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "jupiter-integration-github octokit-rest.js/19.0.5 octokit-core.js/4.1.0 Node.js/18.14.0 (darwin; x64)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.github.com"
            }
          ],
          "headersSize": 812,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "queryString": [],
          "url": "https://api.github.com/app/installations/32841752/access_tokens"
        },
        "response": {
          "bodySize": 475,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 475,
            "text": "{\"token\":\"[REDACTED]\",\"expires_at\":\"2050-12-31T18:09:20Z\",\"permissions\":{\"members\":\"read\",\"organization_administration\":\"read\",\"organization_events\":\"read\",\"organization_secrets\":\"read\",\"actions\":\"read\",\"administration\":\"write\",\"discussions\":\"read\",\"environments\":\"read\",\"issues\":\"read\",\"metadata\":\"read\",\"pages\":\"read\",\"pull_requests\":\"read\",\"secrets\":\"read\",\"security_events\":\"read\",\"vulnerability_alerts\":\"read\"},\"repository_selection\":\"all\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "server",
              "value": "GitHub.com"
            },
            {
              "name": "date",
              "value": "Thu, 09 Feb 2023 23:22:27 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "475"
            },
            {
              "name": "cache-control",
              "value": "public, max-age=60, s-maxage=60"
            },
            {
              "name": "vary",
              "value": "Accept, Accept-Encoding, Accept, X-Requested-With"
            },
            {
              "name": "etag",
              "value": "\"061fb4536f87b2e27070d33c5c72c7147855073de87e7d31602340e5006266c7\""
            },
            {
              "name": "x-github-media-type",
              "value": "github.v3; param=machine-man-preview; format=json"
            },
            {
              "name": "x-github-api-version-selected",
              "value": "2022-11-28"
            },
            {
              "name": "access-control-expose-headers",
              "value": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains; preload"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "referrer-policy",
              "value": "origin-when-cross-origin, strict-origin-when-cross-origin"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none'"
            },
            {
              "name": "x-github-request-id",
              "value": "D578:3282:B9F2DA:17AAF04:63E58033"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 1115,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 201,
          "statusText": "Created"
        },
        "startedDateTime": "2023-02-09T23:22:27.278Z",
        "time": 298,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 298
        }
      },
      {
        "_id": "9ca046c0c3e008801800630bedd7bdf3",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/vnd.github.v3+json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "jupiter-integration-github octokit-rest.js/19.0.5 octokit-core.js/4.1.0 Node.js/18.14.0 (darwin; x64)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.github.com"
            }
          ],
          "headersSize": 761,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.github.com/app/installations/32841752"
        },
        "response": {
          "bodySize": 939,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 939,
            "text": "{\"id\":32841752,\"account\":{\"login\":\"JupiterOne-Sandbox\",\"id\":121981149,\"node_id\":\"O_kgDOB0VI3Q\",\"avatar_url\":\"https://avatars.githubusercontent.com/u/121981149?v=4\",\"gravatar_id\":\"\",\"url\":\"https://api.github.com/users/JupiterOne-Sandbox\",\"html_url\":\"https://github.com/JupiterOne-Sandbox\",\"followers_url\":\"https://api.github.com/users/JupiterOne-Sandbox/followers\",\"following_url\":\"https://api.github.com/users/JupiterOne-Sandbox/following{/other_user}\",\"gists_url\":\"https://api.github.com/users/JupiterOne-Sandbox/gists{/gist_id}\",\"starred_url\":\"https://api.github.com/users/JupiterOne-Sandbox/starred{/owner}{/repo}\",\"subscriptions_url\":\"https://api.github.com/users/JupiterOne-Sandbox/subscriptions\",\"organizations_url\":\"https://api.github.com/users/JupiterOne-Sandbox/orgs\",\"repos_url\":\"https://api.github.com/users/JupiterOne-Sandbox/repos\",\"events_url\":\"https://api.github.com/users/JupiterOne-Sandbox/events{/privacy}\",\"received_events_url\":\"https://api.github.com/users/JupiterOne-Sandbox/received_events\",\"type\":\"Organization\",\"site_admin\":false},\"repository_selection\":\"all\",\"access_tokens_url\":\"https://api.github.com/app/installations/32841752/access_tokens\",\"repositories_url\":\"https://api.github.com/installation/repositories\",\"html_url\":\"https://github.com/organizations/JupiterOne-Sandbox/settings/installations/32841752\",\"app_id\":278393,\"app_slug\":\"j1-ingest-cg\",\"target_id\":121981149,\"target_type\":\"Organization\",\"permissions\":{\"pages\":\"read\",\"issues\":\"read\",\"actions\":\"read\",\"members\":\"read\",\"secrets\":\"read\",\"metadata\":\"read\",\"discussions\":\"read\",\"environments\":\"read\",\"pull_requests\":\"read\",\"administration\":\"write\",\"security_events\":\"read\",\"organization_events\":\"read\",\"organization_secrets\":\"read\",\"vulnerability_alerts\":\"read\",\"organization_administration\":\"read\"},\"events\":[],\"created_at\":\"2023-01-05T16:02:41.000Z\",\"updated_at\":\"2023-01-10T15:35:42.000Z\",\"single_file_name\":null,\"has_multiple_single_files\":false,\"single_file_paths\":[],\"suspended_by\":null,\"suspended_at\":null}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "server",
              "value": "GitHub.com"
            },
            {
              "name": "date",
              "value": "Thu, 09 Feb 2023 23:22:28 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "cache-control",
              "value": "public, max-age=60, s-maxage=60"
            },
            {
              "name": "vary",
              "value": "Accept, Accept-Encoding, Accept, X-Requested-With"
            },
            {
              "name": "etag",
              "value": "W/\"2899e8938fbf66a378063b936eaa7f050e50a593faf1706b9d06e0b595ecda2a\""
            },
            {
              "name": "x-github-media-type",
              "value": "github.v3; format=json"
            },
            {
              "name": "x-github-api-version-selected",
              "value": "2022-11-28"
            },
            {
              "name": "access-control-expose-headers",
              "value": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains; preload"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "referrer-policy",
              "value": "origin-when-cross-origin, strict-origin-when-cross-origin"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none'"
            },
            {
              "name": "x-github-request-id",
              "value": "D589:19DA:A75BFC:155F3AC:63E58034"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 1121,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2023-02-09T23:22:27.604Z",
        "time": 832,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 832
        }
      },
      {
        "_id": "22e4c882794d943e8a6f0961702d5000",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 452,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/vnd.github.v3+json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "jupiterone-graph-github"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "452"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.github.com"
            }
          ],
          "headersSize": 328,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json; charset=utf-8",
            "params": [],
            "text": "{\"query\":\"\\n    query ($login: String!) {\\n      organization(login: $login) {\\n        id\\n        login\\n        name\\n        createdAt\\n        updatedAt\\n        description\\n        email\\n        databaseId\\n        isVerified\\n        location\\n        websiteUrl\\n        url\\n      }\\n      ...on Query {\\n    rateLimit {\\n      limit\\n      cost\\n      remaining\\n      resetAt\\n    }\\n  }\\n    }\",\"variables\":{\"login\":\"JupiterOne-Sandbox\"}}"
          },
          "queryString": [],
          "url": "https://api.github.com/graphql"
        },
        "response": {
          "bodySize": 419,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 419,
            "text": "{\"data\":{\"organization\":{\"id\":\"O_kgDOB0VI3Q\",\"login\":\"JupiterOne-Sandbox\",\"name\":\"JupiterOne\",\"createdAt\":\"2023-01-04T21:30:17Z\",\"updatedAt\":\"2023-01-05T22:32:46Z\",\"description\":\"\",\"email\":\"hello@jupiterone.com\",\"databaseId\":121981149,\"isVerified\":false,\"location\":\"United States of America\",\"websiteUrl\":\"https://jupiterone.com\",\"url\":\"https://github.com/JupiterOne-Sandbox\"},\"rateLimit\":{\"limit\":5000,\"cost\":1,\"remaining\":5000,\"resetAt\":\"2023-02-09T23:27:00Z\"}}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "server",
              "value": "GitHub.com"
            },
            {
              "name": "date",
              "value": "Thu, 09 Feb 2023 23:22:28 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "x-github-media-type",
              "value": "github.v3; format=json"
            },
            {
              "name": "x-ratelimit-limit",
              "value": "5000"
            },
            {
              "name": "x-ratelimit-remaining",
              "value": "5000"
            },
            {
              "name": "x-ratelimit-reset",
              "value": "1675985220"
            },
            {
              "name": "x-ratelimit-used",
              "value": "820"
            },
            {
              "name": "x-ratelimit-resource",
              "value": "graphql"
            },
            {
              "name": "access-control-expose-headers",
              "value": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains; preload"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "referrer-policy",
              "value": "origin-when-cross-origin, strict-origin-when-cross-origin"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none'"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding, Accept, X-Requested-With"
            },
            {
              "name": "x-github-request-id",
              "value": "D58E:5A2E:6FB614:E4C4FD:63E58034"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 1084,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2023-02-09T23:22:28.453Z",
        "time": 296,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 296
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
