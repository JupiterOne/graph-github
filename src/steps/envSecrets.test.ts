import {
  Recording,
  executeStepWithDependencies,
} from '@jupiterone/integration-sdk-testing';
import { buildStepTestConfig } from '../../test/config';
import { setupGithubRecording } from '../../test/recording';
import { Steps } from '../constants';

jest.setTimeout(30000);

let recording: Recording;
afterEach(async () => {
  await recording.stop();
});

test('fetchEnvSecrets exec handler', async () => {
  recording = setupGithubRecording({
    directory: __dirname,
    name: 'envSecrets',
  });

  const stepConfig = buildStepTestConfig(Steps.FETCH_ENV_SECRETS);
  const stepResults = await executeStepWithDependencies(stepConfig);

  expect(stepResults).toMatchStepMetadata(stepConfig);
});
