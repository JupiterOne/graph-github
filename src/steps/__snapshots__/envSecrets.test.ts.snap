// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchEnvSecrets exec handler 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Secret",
      ],
      "_key": "GitHub_Env_devInReporeimagined-barnacle_Secret_DEV_SECRET_1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_at": "2023-01-30T23:27:43Z",
            "name": "DEV_SECRET_1",
            "updated_at": "2023-01-30T23:27:43Z",
          },
        },
      ],
      "_type": "github_env_secret",
      "createdOn": 1675121263000,
      "displayName": "DEV_SECRET_1",
      "name": "DEV_SECRET_1",
      "updatedOn": 1675121263000,
      "visibility": "selected",
      "webLink": "https://github.com/JupiterOne-Sandbox/reimagined-barnacle/settings/environments/835705721/edit",
    },
    Object {
      "_class": Array [
        "Secret",
      ],
      "_key": "GitHub_Env_devInRepoubiquitous-umbrella_Secret_READY_DEV_1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_at": "2023-02-09T22:59:07Z",
            "name": "READY_DEV_1",
            "updated_at": "2023-02-09T22:59:07Z",
          },
        },
      ],
      "_type": "github_env_secret",
      "createdOn": 1675983547000,
      "displayName": "READY_DEV_1",
      "name": "READY_DEV_1",
      "updatedOn": 1675983547000,
      "visibility": "selected",
      "webLink": "https://github.com/JupiterOne-Sandbox/ubiquitous-umbrella/settings/environments/854768849/edit",
    },
    Object {
      "_class": Array [
        "Secret",
      ],
      "_key": "GitHub_Env_prodInRepoubiquitous-umbrella_Secret_READY_PROD_1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_at": "2023-02-09T22:59:52Z",
            "name": "READY_PROD_1",
            "updated_at": "2023-02-09T22:59:52Z",
          },
        },
      ],
      "_type": "github_env_secret",
      "createdOn": 1675983592000,
      "displayName": "READY_PROD_1",
      "name": "READY_PROD_1",
      "updatedOn": 1675983592000,
      "visibility": "selected",
      "webLink": "https://github.com/JupiterOne-Sandbox/ubiquitous-umbrella/settings/environments/854769760/edit",
    },
    Object {
      "_class": Array [
        "Secret",
      ],
      "_key": "GitHub_Env_devInRepoliterate-waffle_Secret_REPO_SEC",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_at": "2023-02-09T23:21:08Z",
            "name": "REPO_SEC",
            "updated_at": "2023-02-09T23:21:08Z",
          },
        },
      ],
      "_type": "github_env_secret",
      "createdOn": 1675984868000,
      "displayName": "REPO_SEC",
      "name": "REPO_SEC",
      "updatedOn": 1675984868000,
      "visibility": "selected",
      "webLink": "https://github.com/JupiterOne-Sandbox/literate-waffle/settings/environments/854794578/edit",
    },
    Object {
      "_class": Array [
        "Secret",
      ],
      "_key": "GitHub_Env_devInRepoliterate-waffle_Secret_SECRET_ONE",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_at": "2023-02-09T23:18:57Z",
            "name": "SECRET_ONE",
            "updated_at": "2023-02-09T23:18:57Z",
          },
        },
      ],
      "_type": "github_env_secret",
      "createdOn": 1675984737000,
      "displayName": "SECRET_ONE",
      "name": "SECRET_ONE",
      "updatedOn": 1675984737000,
      "visibility": "selected",
      "webLink": "https://github.com/JupiterOne-Sandbox/literate-waffle/settings/environments/854794578/edit",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "EN_kwDOI2pL-M4xz9t5",
      "_key": "EN_kwDOI2pL-M4xz9t5|has|GitHub_Env_devInReporeimagined-barnacle_Secret_DEV_SECRET_1",
      "_toEntityKey": "GitHub_Env_devInReporeimagined-barnacle_Secret_DEV_SECRET_1",
      "_type": "github_environment_has_env_secret",
      "displayName": "HAS",
    },
    Object {
      "_class": "USES",
      "_fromEntityKey": "R_kgDOI2pL-A",
      "_key": "R_kgDOI2pL-A|uses|GitHub_Env_devInReporeimagined-barnacle_Secret_DEV_SECRET_1",
      "_toEntityKey": "GitHub_Env_devInReporeimagined-barnacle_Secret_DEV_SECRET_1",
      "_type": "github_repo_uses_env_secret",
      "displayName": "USES",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "EN_kwDOI2pNd84y8rzR",
      "_key": "EN_kwDOI2pNd84y8rzR|has|GitHub_Env_devInRepoubiquitous-umbrella_Secret_READY_DEV_1",
      "_toEntityKey": "GitHub_Env_devInRepoubiquitous-umbrella_Secret_READY_DEV_1",
      "_type": "github_environment_has_env_secret",
      "displayName": "HAS",
    },
    Object {
      "_class": "USES",
      "_fromEntityKey": "R_kgDOI2pNdw",
      "_key": "R_kgDOI2pNdw|uses|GitHub_Env_devInRepoubiquitous-umbrella_Secret_READY_DEV_1",
      "_toEntityKey": "GitHub_Env_devInRepoubiquitous-umbrella_Secret_READY_DEV_1",
      "_type": "github_repo_uses_env_secret",
      "displayName": "USES",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "EN_kwDOI2pNd84y8sBg",
      "_key": "EN_kwDOI2pNd84y8sBg|has|GitHub_Env_prodInRepoubiquitous-umbrella_Secret_READY_PROD_1",
      "_toEntityKey": "GitHub_Env_prodInRepoubiquitous-umbrella_Secret_READY_PROD_1",
      "_type": "github_environment_has_env_secret",
      "displayName": "HAS",
    },
    Object {
      "_class": "USES",
      "_fromEntityKey": "R_kgDOI2pNdw",
      "_key": "R_kgDOI2pNdw|uses|GitHub_Env_prodInRepoubiquitous-umbrella_Secret_READY_PROD_1",
      "_toEntityKey": "GitHub_Env_prodInRepoubiquitous-umbrella_Secret_READY_PROD_1",
      "_type": "github_repo_uses_env_secret",
      "displayName": "USES",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "EN_kwDOI8BHIs4y8yFS",
      "_key": "EN_kwDOI8BHIs4y8yFS|has|GitHub_Env_devInRepoliterate-waffle_Secret_REPO_SEC",
      "_toEntityKey": "GitHub_Env_devInRepoliterate-waffle_Secret_REPO_SEC",
      "_type": "github_environment_has_env_secret",
      "displayName": "HAS",
    },
    Object {
      "_class": "USES",
      "_fromEntityKey": "R_kgDOI8BHIg",
      "_key": "R_kgDOI8BHIg|uses|GitHub_Env_devInRepoliterate-waffle_Secret_REPO_SEC",
      "_toEntityKey": "GitHub_Env_devInRepoliterate-waffle_Secret_REPO_SEC",
      "_type": "github_repo_uses_env_secret",
      "displayName": "USES",
    },
    Object {
      "_class": "OVERRIDES",
      "_fromEntityKey": "GitHub_Env_devInRepoliterate-waffle_Secret_REPO_SEC",
      "_key": "GitHub_Env_devInRepoliterate-waffle_Secret_REPO_SEC|overrides|GitHub_Repo_literate-waffle_Secret_REPO_SEC",
      "_toEntityKey": "GitHub_Repo_literate-waffle_Secret_REPO_SEC",
      "_type": "github_env_secret_overrides_repo_secret",
      "displayName": "OVERRIDES",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "EN_kwDOI8BHIs4y8yFS",
      "_key": "EN_kwDOI8BHIs4y8yFS|has|GitHub_Env_devInRepoliterate-waffle_Secret_SECRET_ONE",
      "_toEntityKey": "GitHub_Env_devInRepoliterate-waffle_Secret_SECRET_ONE",
      "_type": "github_environment_has_env_secret",
      "displayName": "HAS",
    },
    Object {
      "_class": "USES",
      "_fromEntityKey": "R_kgDOI8BHIg",
      "_key": "R_kgDOI8BHIg|uses|GitHub_Env_devInRepoliterate-waffle_Secret_SECRET_ONE",
      "_toEntityKey": "GitHub_Env_devInRepoliterate-waffle_Secret_SECRET_ONE",
      "_type": "github_repo_uses_env_secret",
      "displayName": "USES",
    },
    Object {
      "_class": "OVERRIDES",
      "_fromEntityKey": "GitHub_Env_devInRepoliterate-waffle_Secret_SECRET_ONE",
      "_key": "GitHub_Env_devInRepoliterate-waffle_Secret_SECRET_ONE|overrides|GitHub_Org_JupiterOne-Sandbox_Secret_SECRET_ONE",
      "_toEntityKey": "GitHub_Org_JupiterOne-Sandbox_Secret_SECRET_ONE",
      "_type": "github_env_secret_overrides_org_secret",
      "displayName": "OVERRIDES",
    },
  ],
  "encounteredTypes": Array [
    "github_account",
    "github_repo",
    "github_environment",
    "github_org_secret",
    "github_repo_secret",
    "github_account_owns_repo",
    "github_repo_has_environment",
    "github_account_has_org_secret",
    "github_repo_uses_org_secret",
    "github_repo_has_secret",
    "github_repo_uses_secret",
    "github_repo_secret_overrides_org_secret",
    "github_env_secret",
    "github_environment_has_env_secret",
    "github_repo_uses_env_secret",
    "github_env_secret_overrides_repo_secret",
    "github_env_secret_overrides_org_secret",
  ],
  "numCollectedEntities": 5,
  "numCollectedRelationships": 12,
}
`;
